### PR TITLE
Permit to push measures directly on asset

### PR DESCRIPTION
**Pull request by ChillPC**
#162 

# What does this PR do ?

Permit to push measures directly on asset

## How should this be manually tested?

```sh
curl http://127.0.0.1:7512/_/device-manager/eng1/assets/type1-mod1-ref1/measures --json ' { "measures": [ { "type": "temperature", "values": { "temperature": 99.99 } } ] }'
```

## Other changes

Refinement of service and controller roles

### CODE REVIEW

File: lib/types/measures/Measure.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+   * in the "values" object to get the measure main value.
+   */
+  type: string;
+
+  /**
+   * Property containing the actual measurement.
+   *
+   * This should be specialized by child interfaces
+   */
+  values: JSONObject;
```
[the remark this answers is not in the document]
Comment (ChillPC) [reply lib/types/measures/Measure.ts:23]: `MeasureContent` It was specified as `JSONObject` before, so I did not change that

File: lib/types/measures/Measure.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { JSONObject } from 'kuzzle';
+import { KDocumentContent } from 'kuzzle-sdk';
+
+import { MeasureUnit } from './MeasureDefinition';
+import { MeasureOrigin } from './MeasureOrigin';
+
+/**
+ * Represents a measurement of a value to post on an asset
+ */
+export interface BaseAssetMeasure extends KDocumentContent {
```
Comment (Aschen) [inline lib/types/measures/Measure.ts:10]: As we discussed:
 - a `MeasureContent` represents the content of the entire measure document as it is stored in the database
 - a `Measurement` represents the actual measure 

And finally, the measure name will always be present, not only for the asset (see https://github.com/kuzzleio/kuzzle-device-manager/issues/168#issuecomment-1142182855)

File: lib/utils/measure.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { JSONObject } from 'kuzzle';
+import _ from 'lodash';
+
+export function validateBaseMeasure (toValidate: JSONObject): boolean {
```
Comment (Aschen) [inline lib/utils/measure.ts:4]: Why did you extract this method to use it only once?

**Review by Aschen — CHANGES_REQUESTED** [review]

File: lib/controllers/AssetController.ts [hunk 4]
```diff
@@ -76,6 +84,29 @@ export class AssetController extends CRUDController {
     return { measures };
   }
 
+  async postMeasures (request: KuzzleRequest) {
+    const engineId = request.getString('engineId');
+    request.input.args.index = engineId;
```
[the remark this answers is not in the document]
Comment (ChillPC) [reply lib/controllers/AssetController.ts:89]: Left over from copy paste or something, removed

File: features/AssetController.feature [hunk 5]
```diff
@@ -60,16 +60,32 @@ Feature: Device Manager asset controller
...
       | _source.origin.assetId | _source.origin.id                |
       | "tools-MART-linked"    | "DummyTemp-attached_ayse_linked" |
       | "tools-MART-linked"    | "DummyTemp-attached_ayse_linked" |
       | "tools-MART-linked"    | "DummyTemp-attached_ayse_linked" |
       | "tools-MART-linked"    | "DummyTemp-attached_ayse_linked" |
       | "tools-MART-linked"    | "DummyTemp-attached_ayse_linked" |
 
+
+  Scenario:Add a measure
+    When I successfully execute the action "device-manager/asset":"postMeasures" with args:
```
[the remark this answers is not in the document]
Comment (ChillPC) [reply features/AssetController.feature:78]: Seems like it should be resolve in #168

File: lib/controllers/AssetController.ts [hunk 6]
```diff
@@ -1,27 +1,31 @@
...
   private get sdk () {
     return this.context.accessors.sdk;
   }
 
-  constructor (plugin: Plugin, assetService: AssetService, deviceService : DeviceService) {
+  constructor (
+    plugin: Plugin,
+    assetService: AssetService,
+    deviceService: DeviceService,
+    measureService: MeasureService
```
[the remark this answers is not in the document]
Comment (ChillPC) [reply lib/controllers/AssetController.ts:21]: Too much work for now

**Pull request merged**